merkletree/proof.go

```diff
@@ -65,7 +64,7 @@ func authPathHash(ap *AuthenticationPath,
 
 // VerifyAuthPath should be called after the vrf index is verified successfully
 func VerifyAuthPath(ap *AuthenticationPath,
-	leafIndex, leafCommitment []byte, leafLevel uint32, isLeafEmpty bool,
+	leafIndex []byte, leafCommitment *crypto.Commit, leafLevel uint32, isLeafEmpty bool,
```

Comment:
Could `leafCommitment` just be `[]byte` instead? And in the caller we could pass `leaf.Commitment().Value`, so we don't need a construction step as in https://github.com/coniks-sys/coniks-go/blob/fde67948f80829f9571c8cfb0e52ed90ee7f7532/client/cgo/verification.go#L116 (I feel little awkward when this constructor takes `nil` argument)